### ISSUE TITLE
Cloak of Shadows not working as intended

### ISSUE THREAD

**Issue by yoitspug** [gamefreedomgit/Maelstrom#4780]
[//]: # (REMBEMBER! Add links to things related to the bug using for example:)
[//]: # (http://wowhead.com/)
[//]: # (cata-twinhead.twinstar.cz)

**Description:**
Cloak of Shadows does not fully work as the spell is intended.

**How to reproduce:**
Hop on a Rogue of any spec, level 58 or higher. Place magical effects/DoTs on the rogue and cast Cloak of Shadows to see them being removed. While the spell persists, attempt to reapply magical effects and you will see that the rogue is not immune as he should be.

**How it should work:**
Cloak of Shadows is a level 58 Rogue spell intended to clear all magical effects (any DoT or debuff that isn't physical i.e Faerie Fire, Curse of Agony, Hunter's Mark, etc). It is intended to remove them and keep the rogue immune to re-gaining them for the duration of the spell. As it works now, the spell removes them off the rogue initially but doesn't make the rogue immune to them being reapplied for it's duration. I first noticed this in a BG where a Frost DK applied Frost Fever to me (a magical disease) while Cloak of Shadows persist. The initial cast removed the effects but I did not remain immune to effects being reapplied for the duration of the spell as it was intended. This spell has worked the same way throughout its entire existence, even into current WoW the spell has never changed (been a rogue main for almost a decade now :D) 

**Database links:**
https://cata-twinhead.twinstar.cz/?spell=31224
https://wowwiki-archive.fandom.com/wiki/Cloak_of_Shadows
I couldn't find the WoWHead article but the first link has the OG WoWHead spell description pictured in it

**Comment by Hermeleen** [comment]
this has been talked about.
from an old post on wow freakz:
''As such, any debuffs which are applied by physical abilities will still land on you and still be effective.

This includes, as Kazralar mentioned, Blood Plague and Frost Fever (DK's Plague Strike), along with Censure (any Paladin melee attack), Serpent Sting, Black Arrow and Explosive Shot (Hunter)''

Next, from 8 year old reddit post
"It should be noted that although casting Cloak of Shadows will remove a paladin's Censure debuff, the paladin can reapply it with a Judgment or melee attack despite having cloak up."

**Comment by Sketish** [comment]
Well dk can apply dots during CoS so thats not a bug

**Comment by yoitspug** [comment]
> this has been talked about. from an old post on wow freakz: ''As such, any debuffs which are applied by physical abilities will still land on you and still be effective.
> 
> This includes, as Kazralar mentioned, Blood Plague and Frost Fever (DK's Plague Strike), along with Censure (any Paladin melee attack), Serpent Sting, Black Arrow and Explosive Shot (Hunter)''
> 
> Next, from 8 year old reddit post "It should be noted that although casting Cloak of Shadows will remove a paladin's Censure debuff, the paladin can reapply it with a Judgment or melee attack despite having cloak up."
> 
> I think that what it comes down to is, that cloak of shadows removes all magical debuffs and while the cloak is up, you just have 100% chance to evade those magical debuffs. But, the abilities mentioned above are physical which cloak don't consider as something it should miss.

Respectfully, this just isn't how the spell is intended to operate. Frost Fever, Blood Plague, Black Arrow, Serpent Sting are not physical debuffs. Examples of a physical debuff would be Thunder Clap, Mangle, Sunder Armor, Bleed effects (Rend, Rake, Rip, Hemorrhage). The spell removes all magical debuffs and makes the rogue immune to them for its duration. It's 5 seconds of magical immunity. Spells miss and magical effects don't apply to the rogue. Plague Strike is a melee ability, that does shadow damage and applies a DoT, thus would be ineffective against a Rogue using Cloak of Shadows. Icy Touch/Howling Blast would miss on a rogue under the effects of Cloak of Shadows. Hope this helps! :)

**Comment by Hermeleen** [comment]
> > this has been talked about. from an old post on wow freakz: ''As such, any debuffs which are applied by physical abilities will still land on you and still be effective.
> > This includes, as Kazralar mentioned, Blood Plague and Frost Fever (DK's Plague Strike), along with Censure (any Paladin melee attack), Serpent Sting, Black Arrow and Explosive Shot (Hunter)''
> > Next, from 8 year old reddit post "It should be noted that although casting Cloak of Shadows will remove a paladin's Censure debuff, the paladin can reapply it with a Judgment or melee attack despite having cloak up."
> > I think that what it comes down to is, that cloak of shadows removes all magical debuffs and while the cloak is up, you just have 100% chance to evade those magical debuffs. But, the abilities mentioned above are physical which cloak don't consider as something it should miss.
> 
> Respectfully, this just isn't how the spell is intended to operate. Frost Fever, Blood Plague, Black Arrow, Serpent Sting are not physical debuffs. Examples of a physical debuff would be Thunder Clap, Mangle, Sunder Armor, Bleed effects (Rend, Rake, Rip, Hemorrhage). The spell removes all magical debuffs and makes the rogue immune to them for its duration. It's 5 seconds of magical immunity. Spells miss and magical effects don't apply to the rogue. Plague Strike is a melee ability, that does shadow damage and applies a DoT, thus would be ineffective against a Rogue using Cloak of Shadows. Icy Touch/Howling Blast would miss on a rogue under the effects of Cloak of Shadows. Hope this helps! :)

You are misunderstanding. Of course howling blast is not physical and it does miss when the cloak is up. Of course serpent sti ng is not physical. But the arrow is, and it hits the target, therefore there is no question if the spell missed, because the projectile has nothing to do with magic immunity, and since the projectile hit, the serpent sting gets applied.  To use your thinking, maybe it will help you, imagine that feral rake, would also apply fearie fire, when the cloak is up, rake would still apply it.

**Comment by yoitspug** [comment]
> Well dk can apply dots during CoS so thats not a bug

https://www.youtube.com/watch?v=o3IpdQYWoE8&ab_channel=StaffordJames

I don't think my point is getting across, we're using too much "real world" logic to apply to a fantasy video game. Serpent Sting is a spell, in a real world scenario yes it is an arrow shooting poison but as far as WoW goes it is a magical spell that applies a DoT. It doesn't work against Cloak of Shadows. But to my original point, the reason I made this bug report in the first place, is that DKs were applying DoTs while Cloak of Shadows is up and that is not how the spell works. I went out and found this YouTube video since the Wiki links weren't enough evidence for the team to decree that a bug fix is in order. You can clearly see the Rogue use Cloak of Shadows in his opener and when the DK tries to hit him with Plague Strike while it is active you can clearly see it says "Immune" on his scrolling battle text. Fix the spell. Thanks :)

**Comment by Sketish** [comment]
![image](https://github.com/gamefreedomgit/Maelstrom/assets/122209088/9288c3d6-45bf-4bca-887a-a04eacfd9027)

Some spells have physical instead of magic school thats why they bypass CoS.

**Comment by Hermeleen** [comment]
Although i have to say two things. 1. In the video provided by OP,  the plague strike really immunes on the plague. 2. I been testing some more, and also chains of ice apply diseases. Which again brings me to my point, that you cannot dispel chains of ice slow, because it is physical, and only rets can do that, with cleanse on themselves or freedom on anyone.

**Comment by yoitspug** [comment]
> Although i have to say two things. 1. In the video provided by OP, the plague strike really immunes on the plague. 2. I been testing some more, and also chains of ice apply diseases. Which again brings me to my point, that you cannot dispel chains of ice slow, because it is physical, and only rets can do that, with cleanse on themselves or freedom on anyone. So to repeat myself, Physical Attacks Can Still Apply Magical Dots. Mentioning ret, you keep saying dk, but ret can hit you and apply censure as well. With cloak up.

I keep saying DK because the bug is with DK...

**Comment by yoitspug** [comment]
> ![image](https://user-images.githubusercontent.com/122209088/276540705-9288c3d6-45bf-4bca-887a-a04eacfd9027.png)
> 
> Some spells have physical instead of magic school thats why they bypass CoS.

This is a comment from a random person on a wowhead article... I have provided links to the spell ID stating how they work and video evidence of the spell working as I am describing. I have been playing WoW for over a decade with Cataclysm being the one I was far most proficient in. I'm not trying to be a dick but it really feels like you just don't want to fix the bug or something. The spell isn't working right, please fix it. There is a lot of evidence I have given to support my case, I promise you I know how this spell works...

You are right, some spells have physical schools and I have provided pieces of evidence to support that. Spells that are bleed effects (Rake, Rend, Hemorrhage) or spells that are physical debuffs (Thunder Clap, Sunder Armor, Gouge, Blind) those are spells with physical schools that are not protected against CoS. Plague Strike and Icy Touch/Howling Blast/Chains of Ice do not fall under those catagories.

**Comment by Hermeleen** [comment]
I also beliveve that it should not apply anything magical, when the cloak is up. I tried to talk to chatgpt.. just for fun
he said this:
In the retail version of World of Warcraft during the Cataclysm expansion, Cloak of Shadows was designed to remove and make the Rogue immune to magical debuffs for its duration. This means that diseases, like those applied by Death Knights, should have been removed when Cloak of Shadows was activated, and the Rogue should have been immune to the reapplication of those magical debuffs for the duration of the spell.

The behavior you mentioned in your initial comments, where diseases applied by a Death Knight (e.g., Frost Fever) were removed but could be reapplied despite Cloak of Shadows being active, does not align with how the ability was intended to function in the official Cataclysm version of the game. If this behavior is occurring on a private server, it could be a result of modifications or issues with the server's emulation of the game's mechanics. In the official retail version, Cloak of Shadows should have provided immunity to reapplication of magical debuffs, including diseases.
I am on the same train that it should definitely stop the reapplying of anything while the cloak is up.
